### ISSUE TITLE
Use associative arrays

### ISSUE THREAD

**Issue by matiasjg** [kalimatas/php-liquid#10]
Hi, I am using your PHP Liquid fwk and I made one single change that it could be usefull for others too.
Under the file src/Liquid/Context.php line 240 you have

```
if (preg_match("|\[[0-9]+\]|", $key)) {
  $key = preg_replace("|\[([0-9]+)\]|", ".$1", $key);
}
```

so I changed to:

```
if (preg_match("|\[[0-9]+\]|", $key)) {
  $key = preg_replace("|\[([0-9]+)\]|", ".$1", $key);
} else if (preg_match("|\[[0-9a-z._]+\]|", $key, $matches)) {
  $index = $this->get(str_replace(array("[","]"),"", $matches[0]));
  if ($index) {
    $key = preg_replace("|\[([0-9a-z._]+)\]|", ".$index", $key);
  }
}
```

and now I can use in the liquid template something like this (imagine we have an array with cars assigned to template):

```
{% assign selectedCar = "car-id-1"%}
{% assign myCar = cars[selectedCar] %}

{{ cars.color }}
```

**Comment by matiasjg** [comment]
If you implement this change and you want to use the same example under "for" tags, you have to set the  'ALLOWED_VARIABLE_CHARS' => '[a-zA-Z_.-\[-\]]'

**Comment by kalimatas** [comment]
Hi!

You can make a pull request and I will review it.

Thanks!

**Comment by sanmai** [comment]
Closed in #50
